### PR TITLE
adding lantzproject/lantz package

### PR DESCRIPTION
**Pull request by mtsolmn**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by mtsolmn** [comment]
@conda-forge-admin, please ping team

Hi team,

I am new to writing recipes for conda-forge (or conda-build in general) and have a few questions related to setting up the recipe correctly for this python package.

1. This package is comprised of separate submodules, each of which is in it's on repository. Does the list of source dictionaries download and install in the order listed in the recipe?

2. What does the `-vv` flag in `{{ PYTHON }} -m pip install . -vv` set?  I was unable to find documentation on this.

3. I see in the docs for _Noarch python_ that the `skip: True` directive should not be used with `noarch python`; instead use version constraints. This package supports python3.6 and python>=3.8 (3.7 not supported). I was originally using `skip: true # [py<36 or py==37]`. Is there an equivalent way of writing this using version constraints?

4. This package has a number of python runtime dependencies which are not available on conda-forge. Does this cause a problem? Will conda install these from PYPI using pip if they are available?

5. From the documentation on _Test_ I saw the following,

```
commands:
    - test -f $PREFIX/lib/libboost_log$SHLIB_EXT  # [unix]
    - if not exist %LIBRARY_LIB%\\boost_log-vc140-mt.lib exit 1  # [win]
```
Can this be used without modification? I'm not sure I understand the `$SHLIB_EXT` environment variable.

Any help or clarification would be greatly appreciated. Thank you in advance.

File: recipes/lantz/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  # lantz-core
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}-core/lantz-core-{{ version }}.tar.gz
+    sha256: 9ad1b74c1549fb31db492b3cb836bf710f4b7b9a0c729e0a908c7eec21db0afb
+  # lantz-dev
+  - url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}dev/lantzdev-0.5.2.tar.gz
```
Comment (chrisburr) [inline recipes/lantz/meta.yaml:13]: Why is this version different from core? Should they always be kept in sync? Or is it reasonable to mix and match versions of each component?

**Comment by chrisburr** [comment]
Firstly, I suggest checking out [grayskull](https://github.com/marcelotrevisani/grayskull#grayskull) as it will make adding multiple packages much easier.

> 1. This package is comprised of separate submodules, each of which is in it's on repository. Does the list of source dictionaries download and install in the order listed in the recipe?

I need a little more context to answer this well. See https://github.com/conda-forge/staged-recipes/pull/12502#discussion_r479618438.

> 2. What does the -vv flag in {{ PYTHON }} -m pip install . -vv set? I was unable to find documentation on this.

It increases the verbosity of the output and each additional `v` makes it produce more output. It's useful for debugging when things go wrong so we use `-vv` everywhere.

> 3. I see in the docs for Noarch python that the skip: True directive should not be used with noarch python; instead use version constraints. This package supports python3.6 and python>=3.8 (3.7 not supported). I was originally using skip: true # [py<36 or py==37]. Is there an equivalent way of writing this using version constraints?

Put `python >=3.8` in both `host:` and `run:`.

> 4. This package has a number of python runtime dependencies which are not available on conda-forge. Does this cause a problem? Will conda install these from PYPI using pip if they are available?

You need to add them to conda-forge as well. I would recommend using separate PRs.

> 5. From the documentation on Test I saw the following,

This is for testing packages which provide shared libraries. You don't need to worry about it for Python packages.

> Errors

You probably need to delete `recipes/.DS_Store` though the staged-recipes scripts are not really designed to be ran locally. You might find it easier to test using conda directly using `conda build recipes/my-package/`

**Comment by mtsolmn** [comment]
This is a comment confirming that I am willing to be listed as a maintainer

**Comment by mtsolmn** [comment]
@conda-forge-admin, please ping team

Hi admins,

Thank you for your above comments to help resolve the outstanding issues. I believe everything is ready to be merged. Thanks again.

**Comment by isuruf** [comment]
@mtsolmn, can you respond to https://github.com/conda-forge/staged-recipes/pull/12502#issuecomment-683249288 ?

File: recipes/lantz/meta.yaml (hunk 1 above)
Comment (mtsolmn) [reply recipes/lantz/meta.yaml]: @isuruf The codebase for the project is split between different repositories per the structural choices of the project owner. lantzdev packages all these repos together for publishing and has a different versioning number to denote changes in these packaging requirements.

Comment (isuruf) [reply recipes/lantz/meta.yaml]: I think these should be separate conda packages instead of one conda package.

File: recipes/lantz/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  noarch: python
+  number: 0
+  entry_points:
+    - lantz = lantz.__main__:main
+    - lantz-config = lantz.core.__main__:config
+    - lantz-qtdemo = lantz.qt.__main__:main
+    - lantz-sims = lantz.sims.__main__:main
+    - lantz-ino = lantz.ino.__main__:main
+  script:
+    - {{ PYTHON }} -m pip install https://pypi.io/packages/source/P/PySignal/PySignal-1.1.1.tar.gz
```
Comment (isuruf) [inline recipes/lantz/meta.yaml:44]: Please make a separate conda package for `PySignal`

File: recipes/lantz/meta.yaml (hunk 1 above)
Comment (mtsolmn) [reply recipes/lantz/meta.yaml]: I know it's not intuitive and this is actually something we are working to address on an upcoming release of the software. Lantzdev should really be rolled into Lantz-core as lantzdev is really just a shell. Similarly Lantz-core will not work without lantzdev. So they really shouldn't be packaged separated. I agree this versioning should be simplified and again this is something we are addressing in an upcoming release.

File: recipes/lantz/meta.yaml (hunk 2 above)
Comment (mtsolmn) [reply recipes/lantz/meta.yaml:44]: Please see PR https://github.com/conda-forge/staged-recipes/pull/12557 for this.

Comment (isuruf) [reply recipes/lantz/meta.yaml]: Can you do the same for the other deps?

Comment (mtsolmn) [reply recipes/lantz/meta.yaml]: The remaining deps in the install are all no longer actively maintained. They were written for the original version of this project and are kept for legacy reasons. I don't recommend uploading them as separate packages.

**Comment by mtsolmn** [comment]
@conda-forge-admin, please restart ci

File: recipes/lantz/meta.yaml (hunk 2 above)
Comment (isuruf) [reply recipes/lantz/meta.yaml]: Even if so, it's not our policy to package different packages in the same conda package. Please create separate conda packages.

Comment (mtsolmn) [reply recipes/lantz/meta.yaml]: fixed

**Comment by mtsolmn** [comment]
@conda-forge-admin, please ping team - please merge

**Comment by scopatz** [comment]
Thanks!

**Pull request merged**